src/services/user/hooks/index.js

```diff
@@ -183,72 +192,32 @@ const checkJwt = () => function checkJwtfnc(hook) {
+		return (hasPermission(['STUDENT_CREATE', 'TEACHER_CREATE', 'ADMIN_CREATE'])).call(this, hook);
 	}
 	// eslint-disable-next-line no-underscore-dangle
 	const email = (hook.params._additional || {}).parentEmail || hook.data.email;
 	return hook.app.service('/registrationPins').find({ query: { email, verified: true } })
 		.then((pins) => {
 			if (pins.data.length === 1 && pins.data[0].pin) {
-				const age = globalHooks.getAge(hook.data.birthday);
+				const age = getAge(hook.data.birthday);
 
 				if (!((hook.data.roles || []).includes('student') && age < CONSENT_WITHOUT_PARENTS_MIN_AGE_YEARS)) {
 					hook.app.service('/registrationPins').remove(pins.data[0]._id);
 				}
 
 				return Promise.resolve(hook);
 			}
-			return Promise.reject(new errors.BadRequest('Der Pin wurde noch nicht bei der Registrierung eingetragen.'));
+			return Promise.reject(new BadRequest('Der Pin wurde noch nicht bei der Registrierung eingetragen.'));
 		});
 };
 
-// student administrator helpdesk superhero teacher parent
-// eslint-disable-next-line no-unused-vars
-const permissionRoleCreate = async (hook) => {
-	if (!hook.params.provider) {
-		// internal call
-		return hook;
-	}
-
-	if (hook.data.length <= 0) {
-		return Promise.reject(new errors.BadRequest('No input data.'));
-	}
-
-	let isLoggedIn = false;
-	if (((hook.params || {}).account || {}).userId) {
-		isLoggedIn = true;
-		const userService = hook.app.service('/users/');
-		const currentUser = await userService.get(hook.params.account.userId, { query: { $populate: 'roles' } });
-		const userRoles = currentUser.roles.map((role) => role.name);
-		if (userRoles.includes('superhero')) {
-			// call from superhero Dashboard
-			return Promise.resolve(hook);
-		}
-	}
-
-	if ((isLoggedIn === true && globalHooks.arrayIncludes(
-		(hook.data.roles || []),
-		['student', 'teacher'],
-		['parent', 'administrator', 'helpdesk', 'superhero'],
-	))
-		|| (isLoggedIn === false && globalHooks.arrayIncludes(
-			(hook.data.roles || []),
-			['student', 'parent'],
-			['teacher', 'administrator', 'helpdesk', 'superhero'],
-		))
-	) {
-		return Promise.resolve(hook);
-	}
-	return Promise.reject(new errors.BadRequest('You have not the permissions to create this roles.'));
-};
-
 const securePatching = (hook) => Promise.all([
-	globalHooks.hasRole(hook, hook.params.account.userId, 'superhero'),
-	globalHooks.hasRole(hook, hook.params.account.userId, 'administrator'),
-	globalHooks.hasRole(hook, hook.params.account.userId, 'teacher'),
-	globalHooks.hasRole(hook, hook.params.account.userId, 'demoStudent'),
-	globalHooks.hasRole(hook, hook.params.account.userId, 'demoTeacher'),
-	globalHooks.hasRole(hook, hook.id, 'student'),
+	hasRole(hook, hook.params.account.userId, 'superhero'),
+	hasRole(hook, hook.params.account.userId, 'administrator'),
+	hasRole(hook, hook.params.account.userId, 'teacher'),
+  hasRole(hook, hook.params.account.userId, 'demoStudent'),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 1 tab but found 2 spaces.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5220864)